src/viewer.js

```diff
@@ -668,154 +850,144 @@ class VolumeImageViewer {
+   */
+   getOpticalPathMetadata(opticalPathID) {
+    const channel = this.getOpticalPath(opticalPathID)
+    return channel ? channel.metadata : null;
+  }
 
-    this[_drawingSource] = new VectorSource({
-      tileGrid: tileGrid,
-      projection: projection,
-      features: this[_features],
-      wrapX: false
-    });
+  /** Sets the channel visualization/presentation parameters given an id
+   * @param {string} id of the channel (opticalPathIdentifier)
+   * @param {number[]} color
+   * @param {number} opacity
+   * @param {number[]} thresholdValues
+   * @param {boolean} visible
+   */
+  setOpticalPathBlendingInformation(
+    opticalPathID, 
+    color,
+    opacity,
+    thresholdValues,
+    visible) {
+
+    const channel = this.getOpticalPath(opticalPathID)
+    if (channel === null) {
+      return null;
+    }
 
-    this[_drawingLayer] = new VectorLayer({
-      extent: extent,
-      source: this[_drawingSource],
-      projection: projection,
-      updateWhileAnimating: true,
-      updateWhileInteracting: true,
-    });
+    if (channel.setBlendingInformation(
+        color,
+        opacity,
+        thresholdValues,
+        visible)
+      ) { 
+      this[_map].render();
+    }
+  }
 
-    const view = new View({
-      center: getCenter(extent),
-      extent: extent,
-      projection: projection,
-      resolutions: resolutions,
-      rotation: rotation
-    });
+  /** Gets the channel visualization/presentation parameters given an id
+   * @param {string} opticalPathID of the channel
+   * @param {number[]} color
+   * @param {number} opacity
+   * @param {number[]} thresholdValues
+   * @param {boolean} visible
+   */
+  getOpticalPathBlendingInformation(opticalPathID) {
+    const channel = this.getOpticalPath(opticalPathID)
+    if (channel === null) {
+      return null;
+    }
 
-    this[_interactions] = {
-      draw: undefined,
-      select: undefined,
-      modify: undefined
-    };
+    return channel.getBlendingInformation();
+  }
 
-    this[_controls] = {
-      scale: new ScaleLine({
-        units: 'metric',
-        className: ''
-      })
+  /** Adds the channel to the OpenLayer Map given an id
+   * @param {string} opticalPathID of the channel
+   */
+   activateOpticalPath(opticalPathID) {
+    const channel = this.getOpticalPath(opticalPathID)
+    if (channel === null) {
+      return;
     }
-    if (options.controls.has('fullscreen')) {
-      this[_controls].fullscreen = new FullScreen();
+
+    if (this.isChannelInOpenLayerMap(channel)) {
+      return;
     }
-    if (options.controls.has('overview')) {
-      const overviewImageLayer = new TileLayer({
-        extent: extent,
-        source: rasterSource,
-        preload: 0,
-        projection: projection
-      });
 
-      const overviewView = new View({
-        projection: projection,
-        resolutions: resolutions,
-        rotation: rotation
-      });
+    channel.blendingInformation.addToMap = true;
```

Comment:
Do we need the `addToMap` attribute? Can't we just call `activateOpticalPath()` to add it to the map and `deactivateOpticalPath()` to remove it?

For the constructor, I would just add a subset of optical paths to the map. Maybe even just one to keep the initial network overhead as low as possible and thereby increase the viewer response time (time to first frames being visible to the user).

See also [this comment](https://github.com/MGHComputationalPathology/dicom-microscopy-viewer/pull/47/files#r612396819).

Reply:
yes I removed the addToMap variable, See my comment in https://github.com/MGHComputationalPathology/dicom-microscopy-viewer/pull/47#discussion_r613160177 for the constructor initialization